docs/key-issues/multiple_architectures.md

```diff
@@ -0,0 +1,157 @@
+  and more.
+* Apple transitioned Mac hardware from PowerPC to Intel (x86-64) CPUs, providing
+  a forwards compatibility path for binaries
+* Apple is currently transitioning Mac hardware from Intel (x86-64) to
+  Apple Silicon (ARM64) CPUs, again providing a forwards compatibility
+  path
+* Apple supports ARMv6, ARMv7, ARMv7s, ARM64 and ARM64e on iOS
+* Android currently supports ARMv7, ARM64, x86, and x86-64; it has historically
+  also supported ARMv5 and MIPS
+
+CPU architecture compatibility is a necessary, but not sufficient criterion for
+determining binary compatibility. Even if two binaries are compiled for the same
+CPU architecture, that doesn't guarantee [ABI compatibility](abi.md).
+
+In some respects, CPU architecture compatibility could be considered a superset
+of [GPU compatibility](gpus.md). When dealing with multiple CPU architectures,
+there may be some overal with the solutions that can be used to support GPUs in
+native binaries.
+
+## Platform approaches for dealing with multiple architectures
+
+Three approaches have emerged for handling multiple CPU architectures.
+
+### Multiple binaries
+
+The minimal solution is to distribute multiple binaries. This is the approach
+that was used by Windows NT, and is currently supported by Linux. At time of
+distribution, an installer or other downloadable artefact is provided for each
+supported platform, and it is up to the user to select and download the correct
+artefact.
+
+### Archiving
+
+The approach taken by Android is very similar to the multiple binary approach,
+with some affordances and tooling to simplify distribution.
+
+When building an Android project, each target architecture is compiled
+independently. If a native binary library is required to compile the Android
+application, a version must be provided for each supported CPU architecture. A
+directory layout convention exists for providing a binary for each platform,
+with the same library name. This yields an independent final binary (APK) for
+each CPU architecture. When running locally, a CPU-specific APK will be
+uploaded to the simulator or test device.
+
+This approach can be supported with a conventional "single platform wheel"
+approach. A library developer can package a wheel for each Android CPU
+architecture they wish to support; the Android project will install a
+CPU-architecture appropriate wheel when the compiler pass for that archictecture
+is performed. The only complication is that process of installing wheels will
+involve a dependency resolution pass on each supported platform; this could
+potentially lead to a situation where a single application has different
+versions of a Python library on different architectures.
+
+To simplify the process of distributing the application, at time of publication,
+a single Android App Bundle (AAB) is generated from the multiple CPU-specific
+APKs. This AAB contains binaries for all platforms that can be uploaded to an
+app store.
```

Comment:
It's actually the reverse: the app store generates single-architecture APKs from a multi-architecture AAB, It does this purely to save bandwidth and device storage space: all devices are capable of accepting multi-architecture APKs, and will simply ignore all the architectures other than their own.

For simple apps a multi-architecture APK and an AAB are basically equivalent. But the AAB format has additional support for splitting the app into multiple parts and distributing them independently, e.g. one part per level of a game. But none of this is really relevant to Python, so beyond mentioning the fact that both formats exist, I don't think this page needs to go into any detail about the distinction between them.